File: pipeline/seed.py
"""Seed the Center table from config/centers.json. Idempotent — upserts by slug,
so editing the config and re-running keeps things in sync. (Replaces the old
TS prisma/seed.ts now that the pipeline is Python.)"""
from .config import CENTERS
from .db import get_db


async def run_seed() -> None:
    async with get_db() as db:
        for c in CENTERS:
            fields = {
                "name": c["name"],
                "homepage": c["homepage"],
                "isFocus": c.get("isFocus", False),
                "notes": c.get("notes"),
            }
            await db.center.upsert(
                where={"slug": c["slug"]},
                data={"create": {"slug": c["slug"], **fields}, "update": fields},
            )
            print(f"seeded {c['name']}")

File: prisma/seed.ts
// Seeds the Center table from src/config/centers.ts. Idempotent — upserts by
// slug, so editing the config list and re-running keeps things in sync.
import { PrismaClient } from "@prisma/client";
import { CENTERS } from "../src/config/centers";

const prisma = new PrismaClient();

async function main() {
  for (const c of CENTERS) {
    await prisma.center.upsert({
      where: { slug: c.slug },
      create: {
        slug: c.slug,
        name: c.name,
        homepage: c.homepage,
        isFocus: c.isFocus ?? false,
        notes: c.notes,
      },
      update: {
        name: c.name,
        homepage: c.homepage,
        isFocus: c.isFocus ?? false,
        notes: c.notes,
      },
    });
    console.log(`seeded ${c.name}`);
  }
}

main()
  .then(() => prisma.$disconnect())
  .catch(async (err) => {
    console.error(err);
    await prisma.$disconnect();
    process.exit(1);
  });
